### DISCUSSION TITLE
Redirects not working

### DISCUSSION THREAD

**Discussion by willvlad** [vercel/next.js#21157] [Help]
**What version of Next.js are you using?**

10.0.5

**What version of Node.js are you using?**

12.20.0

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**How are you deploying your application?**

next start

**Describe the Bug**

Redirects not working at all. Here is my next.config.js:
```
const withPlugins = require("next-compose-plugins");
const withImages = require("next-images");
const withSass = require("@zeit/next-sass");
const withCSS = require("@zeit/next-css");
const webpack = require("webpack");
const path = require("path");
require("dotenv").config();

const cssConfig = {
  cssLoaderOptions: {
    url: false,
  },
};

module.exports = withPlugins([[withSass], [withCSS, cssConfig], [withImages]], {
  trailingSlash: true,

//nothing happens here, it still stays on /int-xxx:
  async redirects() {
    return [
      {
        source: "/int-xxx",
        destination: "/international-xxx", 
        permanent: true,
      },
    ];
  },

  webpack(config, { isServer }) {
    const env = Object.keys(process.env).reduce((acc, curr) => {
      acc[`process.env.${curr}`] = JSON.stringify(process.env[curr]);
      return acc;
    }, {});

    if (isServer) {
      require("./scripts/generate-sitemap");
    }

    config.plugins.push(new webpack.DefinePlugin(env));
    config.resolve.modules.push(path.resolve("./"));
    return config;
  },
});

```
Nothing obvious springs out from the above so looks like a bug of redirects implementation. Didn't work in 9.5.0 either.

**Expected Behavior**

Redirects should be working

**To Reproduce**

Run a nextjs site as above

**Comment by thangly98** [comment]
I also have this problem, hope someone can help me

**Comment by zhyd1997** [comment]
Also handled on Stack Overflow: [NextJS redirects not redirecting urls after define in next.config.js file](https://stackoverflow.com/questions/63500859/nextjs-redirects-not-redirecting-urls-after-define-in-next-config-js-file)
